### ISSUE TITLE
cli.test_hammer.test_positive_all_options - needs updating

### ISSUE THREAD

**Issue by rplevka** [SatelliteQE/robottelo#4039]
This might require some confirmation with devs whether all changes have been planned

```bash
E           AssertionError: 
E           hammer activation-key add-host-collection
E             Removed options:
E               * host-collection-organization-id
E           
E           hammer activation-key host-collections
E             Added options:
E               * full-result
E           
E           hammer activation-key list
E             Added options:
E               * full-result
E             Removed options:
E               * full-results
E           
E           hammer activation-key remove-host-collection
E             Removed options:
E               * host-collection-organization-id
E           
E           hammer activation-key subscriptions
E             Added options:
E               * full-result
E             Removed options:
E               * full-results
E           
E           hammer compute-resource image create
E             Added options:
E               * password
E           
E           hammer compute-resource image update
E             Added options:
E               * password
E           
E           hammer content-view add-version
E             Added options:
E               * content-view-version-content-view-id
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer content-view copy
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer content-view delete
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer content-view filter delete
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer content-view filter info
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer content-view filter list
E             Added options:
E               * full-result
E             Removed options:
E               * organization
E               * organization-id
E               * full-results
E               * organization-label
E           
E           hammer content-view filter rule create
E             Added options:
E               * names
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer content-view filter rule delete
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer content-view filter rule info
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer content-view filter rule list
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer content-view filter rule update
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer content-view filter update
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer content-view list
E             Added options:
E               * full-result
E             Removed options:
E               * full-results
E           
E           hammer content-view puppet-module list
E             Added options:
E               * full-result
E             Removed options:
E               * full-results
E           
E           hammer content-view remove
E             Added options:
E               * content-view-version-content-view-ids
E           
E           hammer content-view remove-version
E             Added options:
E               * content-view-version-content-view-id
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer content-view update
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer content-view version incremental-update
E             Added options:
E               * content-view-version-content-view-id
E           
E           hammer content-view version list
E             Added options:
E               * full-result
E             Removed options:
E               * full-results
E           
E           hammer discovery facts
E             Added options:
E               * search
E               * per-page
E               * page
E               * order
E           
E           hammer discovery_rule create
E             Added options:
E               * organization-ids
E               * organizations
E               * location-ids
E               * locations
E           
E           hammer discovery_rule update
E             Added options:
E               * organization-ids
E               * organizations
E               * location-ids
E               * locations
E           
E           hammer docker manifest list
E             Added options:
E               * full-result
E               * content-view-version-content-view-id
E             Removed options:
E               * full-results
E           
E           hammer docker tag list
E             Added options:
E               * full-result
E               * content-view-version-content-view-id
E             Removed options:
E               * full-results
E           
E           hammer erratum list
E             Added options:
E               * full-result
E               * content-view-version-content-view-id
E             Removed options:
E               * full-results
E           
E           hammer filter create
E             Added options:
E               * override
E           
E           hammer filter update
E             Added options:
E               * override
E           
E           hammer gpg list
E             Added options:
E               * full-result
E             Removed options:
E               * full-results
E           
E           hammer host
E             Added subcommands:
E               * rebuild-config
E           
E           hammer host create
E             Added options:
E               * pxe-loader
E           
E           hammer host errata list
E             Added options:
E               * full-result
E             Removed options:
E               * full-results
E           
E           hammer host interface create
E             Added options:
E               * subnet6-id
E               * ip6
E           
E           hammer host interface update
E             Added options:
E               * subnet6-id
E               * ip6
E           
E           hammer host rebuild-config (new command)
E             Added options:
E               * help
E               * name
E               * id
E           
E           hammer host update
E             Added options:
E               * pxe-loader
E           
E           hammer host-collection list
E             Added options:
E               * full-result
E             Removed options:
E               * full-results
E           
E           hammer host-collection update
E             Removed options:
E               * new-organization-id
E           
E           hammer hostgroup create
E             Added options:
E               * pxe-loader
E               * compute-profile
E               * query-organization-id
E               * query-organization-label
E               * query-organization
E               * compute-profile-id
E               * kickstart-repository-id
E             Removed options:
E               * organization
E               * organization-id
E           
E           hammer hostgroup update
E             Added options:
E               * pxe-loader
E               * compute-profile
E               * query-organization-id
E               * query-organization-label
E               * query-organization
E               * compute-profile-id
E               * kickstart-repository-id
E             Removed options:
E               * organization
E               * organization-id
E           
E           hammer location create
E             Added options:
E               * medium-ids
E             Removed options:
E               * media-ids
E           
E           hammer location update
E             Added options:
E               * medium-ids
E             Removed options:
E               * media-ids
E           
E           hammer organization
E             Added subcommands:
E               * add-location
E               * remove-location
E           
E           hammer organization add-location (new command)
E             Added options:
E               * location-id
E               * location
E               * help
E               * name
E               * id
E           
E           hammer organization create
E             Added options:
E               * medium-ids
E             Removed options:
E               * media-ids
E           
E           hammer organization list
E             Added options:
E               * full-result
E             Removed options:
E               * full-results
E           
E           hammer organization remove-location (new command)
E             Added options:
E               * location-id
E               * location
E               * help
E               * name
E               * id
E           
E           hammer organization update
E             Added options:
E               * medium-ids
E             Removed options:
E               * media-ids
E           
E           hammer ostree-branch list
E             Added options:
E               * full-result
E               * content-view-version-content-view-id
E             Removed options:
E               * full-results
E           
E           hammer package list
E             Added options:
E               * full-result
E               * content-view-version-content-view-id
E             Removed options:
E               * full-results
E           
E           hammer package-group list
E             Added options:
E               * full-result
E               * content-view-version-content-view-id
E             Removed options:
E               * full-results
E           
E           hammer product list
E             Added options:
E               * full-result
E             Removed options:
E               * full-results
E           
E           hammer puppet-module list
E             Added options:
E               * full-result
E               * content-view-version-content-view-id
E             Removed options:
E               * full-results
E           
E           hammer realm (new command)
E             Added subcommands:
E               * info
E               * create
E               * list
E               * update
E               * delete
E             Added options:
E               * help
E           
E           hammer realm create (new command)
E             Added options:
E               * organizations
E               * realm-type
E               * name
E               * organization-ids
E               * location-ids
E               * locations
E               * realm-proxy-id
E               * help
E           
E           hammer realm delete (new command)
E             Added options:
E               * help
E               * name
E               * id
E           
E           hammer realm info (new command)
E             Added options:
E               * help
E               * name
E               * id
E           
E           hammer realm list (new command)
E             Added options:
E               * search
E               * help
E               * page
E               * location
E               * organization-id
E               * organization
E               * location-id
E               * order
E               * per-page
E           
E           hammer realm update (new command)
E             Added options:
E               * organizations
E               * realm-type
E               * name
E               * organization-ids
E               * location-ids
E               * locations
E               * realm-proxy-id
E               * id
E               * new-name
E               * help
E           
E           hammer repository create
E             Added options:
E               * verify-ssl-on-sync
E           
E           hammer repository delete
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer repository export
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer repository list
E             Added options:
E               * full-result
E               * ostree-branch-id
E               * content-view-version-content-view-id
E               * ostree-branch
E             Removed options:
E               * full-results
E           
E           hammer repository remove-content
E             Added options:
E               * ids
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E               * content-ids
E           
E           hammer repository update
E             Added options:
E               * verify-ssl-on-sync
E             Removed options:
E               * organization
E               * organization-id
E               * organization-label
E           
E           hammer role create
E             Added options:
E               * organization-ids
E               * organizations
E               * location-ids
E               * locations
E           
E           hammer role update
E             Added options:
E               * organization-ids
E               * organizations
E               * location-ids
E               * locations
E           
E           hammer smart-variable delete
E             Added options:
E               * variable
E           
E           hammer smart-variable info
E             Added options:
E               * variable
E           
E           hammer smart-variable update
E             Added options:
E               * new-variable
E             Removed options:
E               * name
E               * new-name
E           
E           hammer subnet create
E             Added options:
E               * subnet-parameters-attributes
E               * network-type
E           
E           hammer subnet update
E             Added options:
E               * subnet-parameters-attributes
E               * network-type
E           
E           hammer subscription delete-manifest
E             Removed options:
E               * content-host
E               * content-host-id
E           
E           hammer subscription list
E             Added options:
E               * full-result
E             Removed options:
E               * full-results

tests/foreman/cli/test_hammer.py:160: AssertionError
```

**Comment by pondrejk** [comment]
Some regressions identified in above diff
https://bugzilla.redhat.com/show_bug.cgi?id=1416857
https://bugzilla.redhat.com/show_bug.cgi?id=1416862
https://bugzilla.redhat.com/show_bug.cgi?id=1416868

**Comment by pondrejk** [comment]
and
https://bugzilla.redhat.com/show_bug.cgi?id=1426612
http://projects.theforeman.org/issues/18661

**Comment by ntkathole** [comment]
closed via #5748
